### ISSUE TITLE
google_checks.xml: final local variables no violations for non lowerCamelCase

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#20562]
I have read check documentation: https://checkstyle.sourceforge.io/checks/naming/localfinalvariablename.html, https://checkstyle.sourceforge.io/checks/naming/localvariablename.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

[5.2.7 Local Variable Names](https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names)
Local variable names are written in [lowerCamelCase](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case).
Even when final and immutable, local variables are not considered to be constants, and should not be styled as constants.

```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat -n Test.java
     1  /** Some javadoc. */
     2  public class Test {
     3
     4    /** Some javadoc. */
     5    void methodLocalVariable() {
     6      int LocalVariable = 1;              // violation reported
     7      String Variable2 = "";              // violation reported
     8      int VARIABLE = 1;                   // violation reported
     9    }
    10
    11    /** Some javadoc. */
    12    void methodLocalVariableFinal() {
    13      final int LocalVariable = 1;        // violation not reported        
    14      final String Variable2 = "";        // violation not reported
    15      final int VARIABLE = 1;             // violation reported
    16    }
    17  }
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.7.0-all.jar -c /google_checks.xml Test.java
Starting audit...
[WARN] /home/atharv/test/Test.java:6:9: Local variable name 'LocalVariable' must match pattern '^(_|[a-z]([a-z0-9][a-zA-Z0-9]*)?)$'. [LocalVariableName]
[WARN] /home/atharv/test/Test.java:7:12: Local variable name 'Variable2' must match pattern '^(_|[a-z]([a-z0-9][a-zA-Z0-9]*)?)$'. [LocalVariableName]
[WARN] /home/atharv/test/Test.java:8:9: Abbreviation in name 'VARIABLE' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
[WARN] /home/atharv/test/Test.java:8:9: Local variable name 'VARIABLE' must match pattern '^(_|[a-z]([a-z0-9][a-zA-Z0-9]*)?)$'. [LocalVariableName]
[WARN] /home/atharv/test/Test.java:15:15: Abbreviation in name 'VARIABLE' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
Audit done.
```
Expected violations for line 13 `LocalVariable` and line 14 `Variable2`

As a fix https://checkstyle.sourceforge.io/checks/naming/localfinalvariablename.html should be added to google_checks.xml

**Comment by romani** [comment]
fix is merged
